mammoth/trainer.py

```diff
@@ -381,28 +386,40 @@ def validate(self, valid_iter, moving_average=None, task=None):
 
             for batch, metadata, _ in valid_iter:
                 if stats is None:
-                    stats = mammoth.utils.Statistics()
-
-                src, src_lengths = batch.src if isinstance(batch.src, tuple) else (batch.src, None)
-                decoder_input = batch.tgt[:-1]
-                target = batch.tgt[1:]
+                    stats = mammoth.utils.Statistics(n_correct=0)
+
+                stats.n_src_words += batch.src.mask.sum().item()
+                src = batch.src.tensor
+                src_mask = batch.src.mask
+                decoder_input = batch.tgt.tensor[:-1]
+                target = batch.tgt.tensor[1:]
+                # if self.norm_method == "tokens":
+                #     normalization = batch.tgt.mask.sum().item()
+                # else:
+                #     normalization = batch.batch_size
 
                 with torch.cuda.amp.autocast(enabled=self.optim.amp):
                     # F-prop through the model.
                     logits, decoder_output = valid_model(
-                        src,
-                        decoder_input,
-                        src_lengths,
+                        rearrange(src, 't b 1 -> b t'),
```

Comment:
in a perfect world we would just normalize the tensors to batch seq dim across the lib, but this world isn't perfect.

Reply:
Indeed. #80